src/test/regress/expected/grant_on_database_propagation.out

```diff
@@ -0,0 +1,712 @@
+ t
+(1 row)
+
+select has_database_privilege('regression', 'TEMP');
+ has_database_privilege
+---------------------------------------------------------------------
+ t
+(1 row)
+
+select has_database_privilege('regression', 'TEMPORARY');
+ has_database_privilege
+---------------------------------------------------------------------
+ t
+(1 row)
+
+\c - - - :worker_1_port;
+select has_database_privilege('regression', 'CREATE');
+ has_database_privilege
+---------------------------------------------------------------------
+ t
+(1 row)
+
+select has_database_privilege('regression', 'CONNECT');
+ has_database_privilege
+---------------------------------------------------------------------
+ t
+(1 row)
+
+select has_database_privilege('regression', 'TEMP');
+ has_database_privilege
+---------------------------------------------------------------------
+ t
+(1 row)
+
+select has_database_privilege('regression', 'TEMPORARY');
+ has_database_privilege
+---------------------------------------------------------------------
+ t
+(1 row)
+
+\c - - - :master_port
+RESET ROLE;
+revoke CREATE,CONNECT,TEMP,TEMPORARY on database regression from myuser;
+set role myuser;
+select has_database_privilege('regression', 'CREATE');
+ has_database_privilege
+---------------------------------------------------------------------
+ f
+(1 row)
+
+select has_database_privilege('regression', 'CONNECT');
+ has_database_privilege
+---------------------------------------------------------------------
+ f
+(1 row)
+
+select has_database_privilege('regression', 'TEMP');
+ has_database_privilege
+---------------------------------------------------------------------
+ f
+(1 row)
+
+select has_database_privilege('regression', 'TEMPORARY');
+ has_database_privilege
+---------------------------------------------------------------------
+ f
+(1 row)
+
+\c - - - :worker_1_port
+set role myuser;
+select has_database_privilege('regression', 'CREATE');
+ has_database_privilege
+---------------------------------------------------------------------
+ f
+(1 row)
+
+select has_database_privilege('regression', 'CONNECT');
+ has_database_privilege
+---------------------------------------------------------------------
+ f
+(1 row)
+
+select has_database_privilege('regression', 'TEMP');
+ has_database_privilege
+---------------------------------------------------------------------
+ f
+(1 row)
+
+select has_database_privilege('regression', 'TEMPORARY');
+ has_database_privilege
+---------------------------------------------------------------------
+ f
+(1 row)
+
+\c - - - :master_port
+drop user myuser;
+---------------------------------------------------------------------
+-- test CREATE,CONNECT,TEMP,TEMPORARY privileges one by one on database with grant option
+create user myuser;
+create user myuser_1;
+grant CREATE,CONNECT,TEMP,TEMPORARY on database regression to myuser;
+set role myuser;
+grant CREATE,CONNECT,TEMP,TEMPORARY on database regression to myuser_1;
```

Comment:
Let's add a comment here saying that this is expected to be a no-op, because `myuser` was not granted the privileges with `GRANT OPTION`.